terasoluna-gfw-common-libraries/terasoluna-gfw-common/src/main/java/org/terasoluna/gfw/common/codelist/i18n/SimpleI18nCodeList.java

```diff
@@ -226,14 +232,54 @@
 
     /**
      * <p>
-     * returns row of codelist table.
+     * Returns a codelist as map for the {@link org.springframework.context.i18n.LocaleContextHolder}'s locale.
+     * </p>
+     * @see org.terasoluna.gfw.common.codelist.CodeList#asMap()
+     */
+    @Override
+    public Map<String, String> asMap() {
+        return asMap(LocaleContextHolder.getLocale());
+    }
+
+    /**
+     * <p>
+     * returns row of codelist table.<br>
+     * if there is no codelist for the locale, throws an {@link IllegalStateException}.
      * </p>
      * @see org.terasoluna.gfw.common.codelist.i18n.I18nCodeList#asMap(java.util.Locale)
      */
     @Override
     public Map<String, String> asMap(Locale locale) {
         Assert.notNull(locale, "locale is null");
-        return codeListTable.row(locale);
+
+        if (codeListTable.containsRow(locale)) {
+            if (logger.isDebugEnabled()) {
+                logger.debug("Found codelist for specified locale '{}'.",
+                        locale);
+            }
+            return codeListTable.row(locale);
+        }
+
+        Locale langOnlyLocale = new Locale(locale.getLanguage());
+        if (codeListTable.containsRow(langOnlyLocale)) {
+            if (logger.isDebugEnabled()) {
+                logger.debug(
+                        "Found codelist for specified locale '{}' (language only).",
+                        locale);
+            }
+            return codeListTable.row(langOnlyLocale);
+        }
+
+        if (locale.equals(fallbackTo)) {
+            throw new IllegalStateException("No codelist found for specified locale and fail to fallback");
+        }
+
+        if (logger.isDebugEnabled()) {
+            logger.debug(
+                    "There is no codelist for specified locale '{}'. Use '{}' as fallback.",
+                    locale, fallbackTo);
+        }
+        return asMap(fallbackTo);
```

Comment:
Yes, I think we need it.
Because when not set the `fallbackTo` property, use `Locale#getDefault` as default. And it is probable that `Locale#getDefault` returns the locale with country code like "ja_JP".
I think we need to care this case.
What do you think?

Reply:
@btkitsunedukam 
I think...
For a fully fixed default locale, you should define a corresponding code list.
If you want to use default language locale, you should define a fallback locale.

Reply:
@btkitsunedukam 
With https://github.com/terasolunaorg/terasoluna-gfw/pull/840#discussion_r230340541, it is also good to correct the default locale automatically when `fallbackTo` is asserted.

Reply:
Modified by 3136c2b